astroid/bases.py

```diff
@@ -424,13 +420,10 @@ def _infer_type_new_call(self, caller, context):
         for key, value in attrs.items:
             key = next(key.infer(context=context))
             value = next(value.infer(context=context))
-            if key.__class__.__name__ != 'Const':
-                # Something invalid as an attribute.
-                return None
-            if not isinstance(key.value, str):
-                # Not a proper attribute.
-                return None
-            cls_locals[key.value].append(value)
+            if key.__class__.__name__ == 'Const':
```

Comment:
This change will add in `locals` keys that are integers, which is the reason why the original code was there.

Reply:
Hmm ` type.__new__(type, 'a', (), {1:2, "a":5})` is a valid class in `CPython` but returns an error in `pypy`. Do you think we should disallow non string keys regardless?

Reply:
Gotcha, it's great that `pypy` errors on this, since if you do add integers on CPython, then your class will be broken anyway:
```
----> 1 dir(a)

TypeError: '<' not supported between instances of 'str' and 'int'
```

Let's just skip those, so instead of returning, don't add those attributes.